qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -228,34 +261,48 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             if self._sample_from_iterations:
                 power = np.random.randint(power)
             # Run a grover experiment for a given power of the Grover operator.
-            if self._quantum_instance.is_statevector:
-                qc = self.construct_circuit(amplification_problem, power, measurement=False)
-                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+            if self._sampler:
+                qc = self.construct_circuit(amplification_problem, power, measurement=True)
+                job = self._sampler.run([qc])
+                if job.status() is not JobStatus.DONE:
+                    raise QiskitError("The job was not completed successfully. ")
```

Comment:
Is it fine to check the status immediately after submission? I would expect the `JobStatus` is `PENDING` or `RUNNING` for some time after the submission, at least for primitives running on a backend.